rustfst-ffi/src/fst/mod.rs

```diff
@@ -0,0 +1,310 @@
+pub mod concat_fst;
+pub mod const_fst;
+pub mod utils;
+pub mod vector_fst;
+
+use crate::symbol_table::CSymbolTable;
+use crate::tr::CTr;
+use crate::trs::CTrs;
+use crate::{get, get_mut, wrap, CStateId, RUSTFST_FFI_RESULT};
+
+use anyhow::Result;
+use downcast_rs::Downcast;
+use ffi_convert::*;
+use rustfst::algorithms::concat::ConcatFst;
+use rustfst::fst_impls::{ConstFst, VectorFst};
+use rustfst::fst_traits::{Fst, MutableFst, SerializableFst};
+use rustfst::semirings::TropicalWeight;
+use rustfst::Semiring;
+use rustfst::{StateId, SymbolTable, Trs, TrsVec};
+use std::ffi::CStr;
+use std::sync::Arc;
+
+pub trait BindableFst: Downcast {
```

Comment:
We could add a documentation statement for this trait for understanding its purpose: making the binding easier over the variety of possible Fst (LazyFst, etc.)